singer_sdk/plugin_base.py

```diff
@@ -228,19 +228,18 @@ def _validate_config(
         errors: list[str] = []
         log_fn = self.logger.info
         config_jsonschema = self.config_jsonschema
+
         if config_jsonschema:
             self.append_builtin_config(config_jsonschema)
-            try:
-                self.logger.debug(
-                    f"Validating config using jsonschema: {config_jsonschema}"
-                )
-                validator = JSONSchemaValidator(config_jsonschema)
-                validator.validate(self._config)
-            except (ValidationError, SchemaError) as ex:
-                errors.append(str(ex.message))
+            self.logger.debug(
+                f"Validating config using jsonschema: {config_jsonschema}"
+            )
+            validator = JSONSchemaValidator(config_jsonschema)
+            errors = [e.message for e in validator.iter_errors(self._config)]
```

Comment:
These errors may be difficult to read.

Reply:
@WillDaSilva Yup, they're really ugly. I got https://github.com/meltano/sdk/pull/768 for that.